sqldelight-gradle-plugin/src/test/fixtures/case-expression-type/expected/com/test/TestModel.java

```diff
@@ -52,29 +52,29 @@ public T map(@NonNull Cursor cursor) {
     }
   }
 
-  class Marshal<T extends Marshal<T>> {
+  final class Marshal {
     protected ContentValues contentValues = new ContentValues();
 
-    public Marshal() {
+    private Marshal() {
     }
 
-    public Marshal(TestModel copy) {
+    private Marshal(TestModel copy) {
```

Comment:
These should be package scoped or you'll generate a synthetic accessor method when trying to instantiate them from `Factory`.